tests/test_utils.py

```diff
@@ -20,12 +20,70 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import unittest
+import os
+
+try:
+    import boto3
+    import botocore
+    from moto import mock_s3
+except ImportError:
+    boto3 = None
+
+    def mock_s3(cls):
+        """A no-op decorator in case moto mock_s3 can not be imported.
+        """
+        return cls
 
 from lsst.daf.butler.core.utils import iterable, getFullTypeName, Singleton
+from lsst.daf.butler.core.s3utils import (bucketExists, parsePathToUriElements,
+                                          s3CheckFileExists)
 from lsst.daf.butler.core.formatter import Formatter
 from lsst.daf.butler import StorageClass
 
 
+@unittest.skipIf(not boto3, "Warning: boto3 AWS SDK not found!")
```

Comment:
I think I'd prefer this class to be in a `test_s3utils.py` file. The URI tests below I will move to `test_location.py`.

Reply:
Moved, URI tests removed as functionality was removed.